### ISSUE TITLE
Traffic between pods that on the same node

### ISSUE THREAD

**Issue by gaussye** [awsdocs/amazon-eks-user-guide#634]
The guide marks the following sentence as important note: **Pod security group rules aren't applied to traffic between pods or between pods and services, such as kubelet or nodeLocalDNS, that are on the same node.**

Actually I can not understand why Pod security group rules are't applied to traffic between pods. You can not make sure the pods with your customized security group are always not on the same node(I know we can do that with taint, affinity but it make things too complicated). 

I just tested in EKS 1.23 (with only one work node - m5.large), i found the security group can apply to the pods on the same node. Testing scenario:
1. I created 2 pods named pod1 and pod2, go into pod1 and curl ip of pod2 can get response successfully
2. Then created sg-pod1 and sg-pod2 with default setting without inbound rule
3.  Created securitygrouppolicy for pod1(with sg-pod1) and pod2(with sg-pod2).
4. Restart the 2 pods and then go into pod1 and curl ip of pod2 again, I got timeout error which means the security group took effect.
5. Add one inbound rule to sg-pod2 with sg-pod1 as source. Go into pod1 to curl ip of pod2 again can get response successfully.

From my testing result, it proves that the Pod security group rules are applied to traffic between pods on the same node, right?

Would you please advise if there is anything i misunderstand?

**Comment by jdn5126** [comment]
Hi @gaussye , I agree that this is confusingly worded. Security group rules (inbound and outbound) are applied to the resource they are associated with (VPC or EC2 instance). Pod->pod traffic on the same node does not invoke any security group rules because the rules are not associated at the pod-level. Pods in different security groups are unable to communicate because we map each security group to a VLAN on the EC2 instance, and we do not allow routing between VLANs.

In other words, pods in different VLANs cannot communicate, while those in the same VLAN can. You can think of this as the security group being applied at the pod level, but I think the document is trying to clarify that security group rules are only invoked for traffic ingressing/egressing the node.

Hopefully that helps. I am trying to think of a better way to word this so I can raise a PR to update it.
